src/main/java/Tree/CartesianTree.java

```diff
@@ -0,0 +1,79 @@
+package Tree;
+
+import Node.BinaryNode;
+
+
+public class CartesianTree<T> extends BinaryTree<T> {
+    int[] sequence;
+    int length;
+
+    public CartesianTree() { current = null; }
+
+    public CartesianTree(int[] sequence, int length) {
+        current = null;
+        this.sequence = sequence;
+        this.length = length;
+    }
+
+    public void heapTree(Boolean maxHeap) { current = buildHeapTree(this.sequence, 0, length-1, maxHeap); }
+
+    protected BinaryNode<T> buildHeapTree(int[] seq, int begin, int end, Boolean isMaxHeap) {
```

Comment:
and rename this as `constructTree`

Reply:
Also, do we need to to carry seq all the time?
seq is this.sequence and does not change throughout recursion

Reply:
why protected?